src/helpers.rs

```diff
@@ -88,7 +88,17 @@ pub fn load_manifest_path<P: AsRef<Path>>(
     manifest_path: P,
 ) -> Result<Manifest<Meta>, BinstallError> {
     block_in_place(|| {
-        debug!("Reading manifest: {}", manifest_path.as_ref().display());
+        let manifest_path = manifest_path.as_ref();
+        let manifest_path = if manifest_path.is_dir() {
+            manifest_path.join("Cargo.toml")
+        } else {
+            manifest_path.into()
+        };
```

Comment:
Can we also ensure that the `manifest_path` is a file?

```suggestion
        } else if manifest_path.is_file() {
            manifest_path.into()
        } else {
            return Err(BinstallError::CargoManifestPath);
        };
```